### PR TITLE
Introduces ui to manage backend settings

### LINKED ISSUE #1902: Add a ui for the users to manage the json config options
## Problem Definition

While the config works  (on macos I can't locate the directory). It is quite cumbersome to edit the file and find the proper location to put it in different OSes.

We should provide a UI where the user can edit the configuration before logging in.

### PR DESCRIPTION
**Pull request by kelsos**
Closes #1902

### CODE REVIEW

**Comment by kolya182** [comment]
I use `rotki.app` from releases and build docker image from `develop` and `bug-fixes` branches for latest features and it's good to point both apps to the same db. And also will be  good to have option to locate/point rotki db on rotki.app first start.

**Comment by kelsos** [comment]
@kolya182 first at least when it comes to development, you should not use the same database in both production/bug fixes, this might create issues.
Second I am not sure how easy would be to share between the app/docker because I think docker requires the data-dir to have root as the owner.

**Comment by kolya182** [comment]
> @kolya182 first at least when it comes to development, you should not use the same database in both production/bug fixes, this might create issues.
> Second I am not sure how easy would be to share between the app/docker because I think docker requires the data-dir to have root as the owner.

Can't wait to test it.

**Comment by LefterisJP** [comment]
@kolya182 please don't use develop. It's supposed to be an unstable branch and it will most probably always have issues. If you do, always take backups of your DBs.

File: docs/contribute.rst [hunk 1]
```diff
@@ -20,13 +20,13 @@ Run Rotki in debug mode
 For running Rotki in debug mode, you can do it either via a config file or the app UI:
 
 - **Config file**: see the section :ref:`set-the-backend-s-arguments`.
-- **App UI**: before log in, click the cog wheel at the bottom right corner and select "Debug" (image below). Then log in as usual.
+- **App UI**: before log in, click the cog wheel at the bottom right corner and select "Debug" (image below). Pres the save button and proceed to log in as usual.
```
Comment (LefterisJP) [inline docs/contribute.rst:23]:
```suggestion
- **App UI**: before log in, click the cog wheel at the bottom right corner and select "Debug" (image below). Press the save button and proceed to log in as usual.
```

**Comment by kolya182** [comment]
> @kolya182 please don't use develop. It's supposed to be an unstable branch and it will most probably always have issues. If you do, always take backups of your DBs.

Hope this new UI will fix issue with picking db location on user creation

**Review by LefterisJP — COMMENTED** [review]
All right so seems to work but there is some problems/things need fixing.

### Advanced

Make some options hidden under an advanced tab, or at least separate them in an advanced section



### Replace the screenshots in the docs

The docs section you are touching have screenshots.  For example the `images/rotki_debug_mode_set.png`. Replace them with the new ones.

Also mention the ability that users have to change the backend settings in the docs and add a screenshot.

### Data directory resets without pressing reset to defaults

Some times, the data directory seems to reset to default when playing with the settings, without confirming a "reset to default".

I reproduced it multiple times through different paths.

One was:

1. Start with custom data and logs directory.
2. Change the logs again to another directory.
3. Press reset to defaults, BUT then press cancel and not confirm.
4. Then press save.

The new logs directory will be there but the data directory will be reset to default. Sent you a video in a DM about it.

**Review by LefterisJP — COMMENTED** [review]
Previous stuff is fixed. Few more comments.

### Starting while the directory is deleted

I started the branch again and it remembered my previous directory. I had deleted that directory though, so it is stuck at connecting since it can't find it:

```
2021-05-18T21:56:57.273Z: Restarting the backend                   
2021-05-18T21:56:57.279Z: Determining if we are packaged by seeing if /home/lefteris/w/rotkehlchen/frontend/app/node_modules/electron/dist/resources/rotkehlchen_py_dist exists                                                                                                  
Error: ENOENT: no such file or directory, open '/home/lefteris/DELMELOGS2/rotki_electron.log'                                           
    at Object.openSync (fs.js:476:3)                                                                                                    
    at Object.func [as openSync] (electron/js2c/asar_bundle.js:5:1846)                                                                                                                                                                                                           
    at Object.writeFileSync (fs.js:1467:35)                                                                                                                                                                                                                                      
    at Object.appendFileSync (fs.js:1506:6)                                                                                                                                                                                                                                          at PyHandler.logToFile (webpack:///./src/py-handler.ts?:114:58)                                                                                                                                                                                                              
    at PyHandler.guessPackaged (webpack:///./src/py-handler.ts?:229:14)                                                                                                                                                                                                          
    at PyHandler.createPyProc (webpack:///./src/py-handler.ts?:169:18)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async IpcMainImpl.eval (webpack:///./src/electron-main/ipc-setup.ts?:57:13) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/home/lefteris/DELMELOGS2/rotki_electron.log'
}
```

So either fail with a proper error if the directory is not found OR just create it.

### Can't terminate app

When the timeout timer ran out and I got the terminate app choice I tried to press it to terminate it and nothing happens. I am not sure if this is actually related to this PR but mentioning it here since I experienced it.

Electron log:

```
(node:30508) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/home/lefteris/DELMELOGS2/rotki_electron.log'
    at Object.openSync (fs.js:476:3)
    at Object.func [as openSync] (electron/js2c/asar_bundle.js:5:1846)
    at Object.writeFileSync (fs.js:1467:35)
    at Object.appendFileSync (fs.js:1506:6)
    at PyHandler.logToFile (webpack:///./src/py-handler.ts?:114:58)
    at PyHandler.exitPyProc (webpack:///./src/py-handler.ts?:208:14)
    at closeApp (webpack:///./src/background.ts?:124:21)
    at IpcMainImpl.eval (webpack:///./src/electron-main/ipc-setup.ts?:98:150)
    at IpcMainImpl.emit (events.js:315:20)
    at Object.<anonymous> (electron/js2c/browser_init.js:161:9692)
(node:30508) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 34)
2021-05-18T21:57:48.082Z: Terminating the backend
(node:30508) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/home/lefteris/DELMELOGS2/rotki_electron.log'
    at Object.openSync (fs.js:476:3)
    at Object.func [as openSync] (electron/js2c/asar_bundle.js:5:1846)
    at Object.writeFileSync (fs.js:1467:35)
    at Object.appendFileSync (fs.js:1506:6)
    at PyHandler.logToFile (webpack:///./src/py-handler.ts?:114:58)
    at PyHandler.exitPyProc (webpack:///./src/py-handler.ts?:208:14)
    at closeApp (webpack:///./src/background.ts?:124:21)
    at IpcMainImpl.eval (webpack:///./src/electron-main/ipc-setup.ts?:98:150)
    at IpcMainImpl.emit (events.js:315:20)
    at Object.<anonymous> (electron/js2c/browser_init.js:161:9692)
(node:30508) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 36)
2021-05-18T21:57:48.274Z: Terminating the backend
(node:30508) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/home/lefteris/DELMELOGS2/rotki_electron.log'
    at Object.openSync (fs.js:476:3)
    at Object.func [as openSync] (electron/js2c/asar_bundle.js:5:1846)
    at Object.writeFileSync (fs.js:1467:35)
    at Object.appendFileSync (fs.js:1506:6)
    at PyHandler.logToFile (webpack:///./src/py-handler.ts?:114:58)
    at PyHandler.exitPyProc (webpack:///./src/py-handler.ts?:208:14)
    at closeApp (webpack:///./src/background.ts?:124:21)
    at IpcMainImpl.eval (webpack:///./src/electron-main/ipc-setup.ts?:98:150)
    at IpcMainImpl.emit (events.js:315:20)
    at Object.<anonymous> (electron/js2c/browser_init.js:161:9692)
(node:30508) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 38)
2021-05-18T21:57:55.707Z: Terminating the backend
(node:30508) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/home/lefteris/DELMELOGS2/rotki_electron.log'
    at Object.openSync (fs.js:476:3)
    at Object.func [as openSync] (electron/js2c/asar_bundle.js:5:1846)
    at Object.writeFileSync (fs.js:1467:35)
    at Object.appendFileSync (fs.js:1506:6)
    at PyHandler.logToFile (webpack:///./src/py-handler.ts?:114:58)
    at PyHandler.exitPyProc (webpack:///./src/py-handler.ts?:208:14)
    at closeApp (webpack:///./src/background.ts?:124:21)
    at IpcMainImpl.eval (webpack:///./src/electron-main/ipc-setup.ts?:98:150)
    at IpcMainImpl.emit (events.js:315:20)
    at Object.<anonymous> (electron/js2c/browser_init.js:161:9692)
(node:30508) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 40)
2021-05-18T21:58:03.070Z: Terminating the backend
(node:30508) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/home/lefteris/DELMELOGS2/rotki_electron.log'
    at Object.openSync (fs.js:476:3)
    at Object.func [as openSync] (electron/js2c/asar_bundle.js:5:1846)
    at Object.writeFileSync (fs.js:1467:35)
    at Object.appendFileSync (fs.js:1506:6)
    at PyHandler.logToFile (webpack:///./src/py-handler.ts?:114:58)
    at PyHandler.exitPyProc (webpack:///./src/py-handler.ts?:208:14)
    at closeApp (webpack:///./src/background.ts?:124:21)
    at App.eval (webpack:///./src/background.ts?:73:15)
    at App.emit (events.js:315:20)
    at App.eval (webpack:///./src/background.ts?:68:125)
(node:30508) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 42)
```

File: docs/contribute.rst [hunk 2]
```diff
@@ -20,13 +20,13 @@ Run Rotki in debug mode
 For running Rotki in debug mode, you can do it either via a config file or the app UI:
 
 - **Config file**: see the section :ref:`set-the-backend-s-arguments`.
-- **App UI**: before log in, click the cog wheel at the bottom right corner and select "Debug" (image below). Then log in as usual.
+- **App UI**: before log in, click the cog wheel at the bottom right corner and select "Debug" (image below). Press the save button and proceed to log in as usual.
 
 .. image:: images/rotki_debug_mode_set.png
```
[the remark this answers is not in the document]
Comment (kelsos) [reply docs/contribute.rst:25]: Do you mean this screenshot?
image

[the remark this answers is not in the document]
Comment (kelsos) [reply docs/contribute.rst:25]: The screenshot was first changed with the initial modifications, along with the doc changes, and then updated to the recent changes. For both comments they were in the diff so I am not sure why I am getting constant reminders to change the screeshot.

[the remark this answers is not in the document]
Comment (LefterisJP) [reply docs/contribute.rst:25]: Ah yea okay sorry. I somehow expected to see a change in the diff of the docs. :rofl:

**Comment by kelsos** [comment]
It seems to be related to the development server

**Comment by kelsos** [comment]
So this restarting problem was related to the log directory errors. (since it was trying to write a file that didn't exist)

**Review by LefterisJP — COMMENTED** [review]
Hey @kelsos didn't we say that we would create the directory if it's not found.

I opened the branch again and it tries to open the "DELME3" directory which was deleted and it fails like this:




Terminate also does not work and the settings button does not appear. So I can't even choose to reset the setting.

**Review by LefterisJP — APPROVED** [review]
Actually

> Hey @kelsos didn't we say that we would create the directory if it's not found.

I realized this was a backend error. And seems we were not creating that directory if it did not exist. So I added an one liner to do so.

Other stuff seem to work.

**Review by LefterisJP — APPROVED** [review]
The only failed backend test is due to Kusama 409 hiccup:

```
2021-05-19T23:04:09.4090662Z [19/05/2021 22:58:13 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": null, "message": "Kusama failed to connect to own node at endpoint https://kusama.api.onfinality.io/public-https, due to Kusama failed to request last block at endpoint: https://kusama.api.onfinality.io/public-https due to: Kusama node failed to request last block. Returned None.."}, status_code=409
```

So will merge

**Pull request merged**